### DISCUSSION TITLE
pdm sync vs pdm install --frozen-lockfile

### DISCUSSION THREAD

**Discussion by deronnax** [pdm-project/pdm#2761] [Q&A] [answered]
Isn't `pdm install --frozen-lockfile` strictily equivalent to pdm sync? And if so, should not we remove `--frozen-lockfile` from `sync`, to avoid to have 2 ways of doing the same thing?

**Comment by frostming** [comment, marked as answer]
`pdm install --frozen-lockfile` may still **update** the pinned versions, but the updates won't be reflected in the lockfile. While `pdm sync` is guaranteed to be a lock-free operation that installs whatever in the lockfile.

You can try it yourself by doing some modifications to the `pyproject.toml` and run those commands

**Reply by deronnax** [reply]
thank you very much 🙏
